lib/sidekiq/scheduled.rb

```diff
@@ -39,33 +38,57 @@ def enqueue_jobs(now=Time.now.to_f.to_s, sorted_sets=SETS)
     # workers can pick it up like any other job.
     class Poller
       include Util
-      include Actor
 
       INITIAL_WAIT = 10
 
       def initialize
         @enq = (Sidekiq.options[:scheduled_enq] || Sidekiq::Scheduled::Enq).new
+        @done = false
+        @queue = ConnectionPool::TimedStack.new
       end
 
-      def poll(first_time=false)
-        watchdog('scheduling poller thread died!') do
-          initial_wait if first_time
-
-          begin
-            @enq.enqueue_jobs
-          rescue => ex
-            # Most likely a problem with redis networking.
-            # Punt and try again at the next interval
-            logger.error ex.message
-            logger.error ex.backtrace.first
+      # Shut down this Fetcher instance, will pause until
+      # the thread is dead.
+      def terminate
+        @done = true
+        if @thread
+          t = @thread
+          @thread = nil
+          @queue << 0
+          t.value
+        end
+      end
+
+      def start
+        @thread ||= safe_thread("scheduler") do
```

Comment:
Presumably the allocated thread will just sleep while the queue is empty, so it might be possible to just make `start` private and call it from `initialize`. Then you can eliminate the conditional here and in `terminate`. </armchairengineering>